replications/internal/queue_management_test.go

```diff
@@ -5,96 +5,247 @@ import (
+	require.NoDirExists(t, filepath.Join(enginePath, "replicationq", id1.String()))
 }
 
 func TestDeleteQueueNonexistentID(t *testing.T) {
 	t.Parallel()
 
-	tempEnginePath, err := os.MkdirTemp("", "engine")
-	require.NoError(t, err)
-	defer os.RemoveAll(tempEnginePath)
-
-	logger := zaptest.NewLogger(t)
-	qm := NewDurableQueueManager(logger, tempEnginePath)
+	enginePath, qm := initQueueManager(t)
+	defer os.RemoveAll(enginePath)
 
 	// Delete nonexistent queue
-	err = qm.DeleteQueue(replicationID)
+	err := qm.DeleteQueue(id1)
 	require.EqualError(t, err, "durable queue not found for replication ID \"0000000000000001\"")
 }
 
 func TestUpdateMaxQueueSizeNonexistentID(t *testing.T) {
 	t.Parallel()
 
-	tempEnginePath, err := os.MkdirTemp("", "engine")
+	enginePath, qm := initQueueManager(t)
+	defer os.RemoveAll(enginePath)
+
+	// Update nonexistent queue
+	err := qm.UpdateMaxQueueSize(id1, influxdb.DefaultReplicationMaxQueueSizeBytes)
+	require.EqualError(t, err, "durable queue not found for replication ID \"0000000000000001\"")
+}
+
+func TestStartReplicationQueue(t *testing.T) {
+	t.Parallel()
+
+	enginePath, qm := initQueueManager(t)
+	defer os.RemoveAll(enginePath)
+
+	// Create new queue
+	err := qm.InitializeQueue(id1, maxQueueSizeBytes)
+	require.NoError(t, err)
+	require.DirExists(t, filepath.Join(enginePath, "replicationq", id1.String()))
+
+	// Represents the replications tracked in sqlite, this one is tracked
+	trackedReplications := make(map[platform.ID]int64)
+	trackedReplications[id1] = maxQueueSizeBytes
+
+	// Simulate server shutdown by closing all queues and clearing replicationQueues map
+	shutdown(t, qm)
+
+	// Call startup function
+	err = qm.StartReplicationQueues(trackedReplications)
+	require.NoError(t, err)
+
+	// Make sure queue is stored in map
+	require.NotNil(t, qm.replicationQueues[id1])
+
+	// Ensure queue is open by trying to remove, will error if open
+	err = qm.replicationQueues[id1].Remove()
+	require.Errorf(t, err, "queue is open")
+}
+
+func TestStartReplicationQueuePartialDelete(t *testing.T) {
+	t.Parallel()
+
+	enginePath, qm := initQueueManager(t)
+	defer os.RemoveAll(enginePath)
+
+	// Create new queue
+	err := qm.InitializeQueue(id1, maxQueueSizeBytes)
+	require.NoError(t, err)
+	require.DirExists(t, filepath.Join(enginePath, "replicationq", id1.String()))
+
+	// Represents the replications tracked in sqlite, replication above is not tracked (not present in map)
+	trackedReplications := make(map[platform.ID]int64)
+
+	// Simulate server shutdown by closing all queues and clearing replicationQueues map
+	shutdown(t, qm)
+
+	// Call startup function
+	err = qm.StartReplicationQueues(trackedReplications)
+	require.NoError(t, err)
+
+	// Make sure queue is not stored in map
+	require.Nil(t, qm.replicationQueues[id1])
+
+	// Check for queue on disk, should be removed
+	require.NoDirExists(t, filepath.Join(enginePath, id1.String()))
```

Comment:
I think the `filepath.Join` here is missing `replicationq`, so the check will always pass. Refactoring so the test doesn't need to pass `replicationq` would be a better fix than adding the hard-coded string IMO